apps/desktop-wallet/src/components/NFTCard.tsx

```diff
@@ -16,22 +16,24 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { NFT, selectNFTById } from '@alephium/shared'
+import { NFT } from '@alephium/shared'
 import { colord } from 'colord'
 import { motion } from 'framer-motion'
 import styled from 'styled-components'
 
+import { useAddressesNfts } from '@/api/addressesNftsDataHooks'
 import NFTThumbnail from '@/components/NFTThumbnail'
 import Truncate from '@/components/Truncate'
-import { useAppSelector } from '@/hooks/redux'
 
 interface NFTCardProps {
   nftId: NFT['id']
   onClick?: () => void
 }
 
 const NFTCard = ({ nftId, onClick }: NFTCardProps) => {
-  const nft = useAppSelector((s) => selectNFTById(s, nftId))
+  const { data: nfts } = useAddressesNfts()
+
+  const nft = nfts.find((nft) => nft.id === nftId)
```

Comment:
This doesn't look like a very smart way of getting the NFT data, but with the current setup it works. Maybe we can return a `Record<NftId, NFT>` instead of `NFT[]` from the `useAddressesNfts`. To be looked into during optimization phase.